tensorboard/plugins/hparams/hparams_plugin.py

```diff
@@ -191,13 +191,21 @@ def _get_scalars_plugin(self):
 
 
 def _parse_request_argument(request, proto_class):
-    if request.method == "POST":
-        return json_format.Parse(request.data, proto_class())
-
-    # args.get() returns the request URI-unescaped.
-    request_json = request.args.get("request")
+    request_json = (
+        request.data
+        if request.method == "POST"
+        else request.args.get("request")
+    )
     if request_json is None:
         raise error.HParamsError(
-            "Expected a JSON-formatted 'request' arg of type: %s" % proto_class
+            "Expected a JSON-formatted request data of type: %s" % proto_class
+        )
+    try:
+        return json_format.Parse(request_json, proto_class())
+    except json_format.ParseError as e:
+        err_desc = (
+            "Failed to parse the request json {} into a proto: {}".format(
+                request_json, str(e)
+            )
         )
-    return json_format.Parse(request_json, proto_class())
+        raise werkzeug.exceptions.BadRequest(description=err_desc)
```

Comment:
I just realized that the code calling this method would catch an HParamsError and convert it to a BadRequest error, so let's be consistent and make this an HParamsError as well. Sorry for not catching this earlier.